core/templates/pages/exploration-player-page/learner-experience/conversation-skin.component.ts

```diff
@@ -1060,22 +1061,20 @@ export class ConversationSkinComponent {
   }
 
   triggerIfLearnerStuckAction(): void {
-    if (this.timeout) {
-      clearTimeout(this.timeout);
-      this.timeout = null;
-    }
     if (this.responseTimeout) {
       clearTimeout(this.responseTimeout);
       this.responseTimeout = null;
     }
     this.responseTimeout = setTimeout(() => {
-      if (this.nextCardIfStuck) {
+      if (this.nextCardIfStuck && this.nextCardIfStuck !== this.displayedCard) {
         // Let the learner know about the redirection to a state
         // for clearing concepts.
         this.playerTranscriptService.addNewResponseToExistingFeedback(
           this.translateService.instant(
             'I18N_REDIRECTION_TO_STUCK_STATE_MESSAGE')
         );
+        // Enable visibility of ContinueStuck button.
```

Comment:
```suggestion
        // Enable visibility of ContinueToRevise button.
```
Change for all usages.

Reply:
Done.